CONTRIBUTING.md

```diff
@@ -176,64 +189,97 @@ It is encouraged before submission to rebase your changes on top of master and t
 The VireoSDK directory will be symlinked to the `node_modules` folder of your application.
 This allows you to make changes to your local vireo and have them be instantly available to the consuming application.
 
 **WARNING** Since the local VireoSDK is symlinked to the JavaScript application be aware of tooling that could accidently delete or modify your local development files unexpectedly.
 
-### Testing on a different machine / CI
+### Testing vireo.js on a different machine / CI
+
 If you want to test your local vireo changes on a separate machine from your local machine (a CI or test machine) without publishing to the npm registry you can do the following:
-- In your local VireoSDK directory run: `npm pack`
-- This will create a `vireo-<VERSION>.tgz` file
-- Move the tgz file to a location that can be bundled with your app. A good location would be next to the package.json and package-lock.json file
-- With the package.json and package-lock.json files checked out / editable run:
+
+1. In your local VireoSDK directory run: `npm pack`
+2. This will create a `vireo-<VERSION>.tgz` file
+3. Move the tgz file to a location that can be bundled with your app. A good location would be next to the package.json and package-lock.json file
+4. With the package.json and package-lock.json files checked out / editable run:
+
   ```console
   npm install <PATH_TO_TGZ_FILE>
   ```
 
 This will update the package.json and package-lock.json to point to the tgz file location.
 You can now share the application or if running in a CI environment create a changeset, etc, that has the tgz file and updated package.json and package-lock.json to test.
 
+## Testing esh.exe in a .NET application
+
+The esh.exe build is packaged as a nuget package to make it consumeable in .NET applications. There are two primary workflows for testing your local Vireo changes in other .NET based applications:
+
+### Testing esh.exe on your local machine
+
+1. If you do not already have nuget.exe, download it [here](https://www.nuget.org/downloads). Version 4.7.1 should work.
+2. In the directory with the `VireoSDK.nuspec` file run:
+
+    ```console
+    nuget pack VireoSDK.nuspec -properties version="[version]"
+    ```
+    replacing `[version]` with some version.
+
+    This command will generate a `.nupkg` file in the folder the command is run from.
+3. See [installing a nuget package](https://docs.microsoft.com/en-us/nuget/consume-packages/ways-to-install-a-package) for the various ways to install the created package to your .NET application.
+
+### Testing esh.exe on a different machine / CI
+
+1. Follow the same steps above for testing on your local machine.
+2. Once you have installed the nuget package, you must add a package source for that package so the remote machine can download the package. See [publishing a package](https://docs.microsoft.com/en-us/nuget/create-packages/publish-a-package) for information how to do this.
```

Comment:
Is it not possible to reference a nupkg file that you add to the shelveset for testing? Requiring the package be published for testing is a bit extreme.

Reply:
I see what you mean. At least with visual studio, the sources for a nuget are defined in the NuGet.config file (https://docs.microsoft.com/en-us/nuget/reference/nuget-config-file). Currently we have nuget.org and the LabVIEW TFS packages referenced here, but that would require we publish to one of those feeds. I thought I could add a reference to the "Tests.HtmlVI" project, but then I realized that the path is absolute in the examples and I don't know what the path would be on the CI machine.

We can talk about this in person if you would like. My plan for our squad is to just install the package locally on your machine, and then add the esh.exe directly to the shelveset instead of trying to reference some local package. Or, perhaps we could write a script that publishes a test package to our internal ni feed so as to not pollute the public nuget.org vireo feed.

As for this document though, I wanted to be as general as possible so others would know how to actually publish a test package if that's what they wanted to do.

Reply:
I found another way to do this without having to publishing to a feed. Also, there is a way to reference relative paths in the NuGet.config file.